src/backends/torch/optim/ranger.h

```diff
@@ -0,0 +1,107 @@
+#pragma GCC diagnostic push
+#pragma GCC diagnostic ignored "-Wunused-parameter"
+#include <torch/arg.h>
+#include <torch/nn/module.h>
+#include <torch/optim/optimizer.h>
+#include <torch/optim/serialize.h>
+#pragma GCC diagnostic pop
+
+#include <utility>
+#include <vector>
+
+namespace torch
+{
+  namespace serialize
+  {
+    class OutputArchive;
+    class InputArchive;
+  } // namespace serialize
+} // namespace torch
+
+namespace dd
+{
+  struct TORCH_API RangerOptions
+      : public torch::optim::OptimizerCloneableOptions<RangerOptions>
+  {
+    RangerOptions(double lr = 1e-3);
+    TORCH_ARG(double, lr) = 1e-3;
+    typedef std::tuple<double, double> betas_t;
+    TORCH_ARG(betas_t, betas) = std::make_tuple(0.9, 0.999);
+    TORCH_ARG(double, eps) = 1e-8;
+    TORCH_ARG(double, weight_decay) = 0.0;
```

Comment:
Just double checking: are we sure `weight_decay` is set to 0.0 by default ?

Reply:
default is no weight_decay in almost all optimizer implementation I've seen

Reply:
OK, j etais sur que c etait active partout.. comme quoi...